src/commands/plugins/uninstall.ts

```diff
@@ -64,15 +66,22 @@ export default class PluginsUninstall extends Command {
 
       try {
         const {name} = unfriendly
-        const displayName = friendly === '.' ? name : friendly ?? name
+        const displayName = friendly === '.' ? name : (friendly ?? name)
         ux.action.start(`${this.config.name}: Uninstalling ${displayName}`)
         await plugins.uninstall(name)
+        pluginNameToDelete.push(name)
       } catch (error) {
         ux.action.stop(bold.red('failed'))
         throw error
       }
 
       ux.action.stop()
     }
+
+    for (const p of pluginNameToDelete) {
+      this.config.plugins.delete(p)
+    }
```

Comment:
I know we do the same here:
https://github.com/oclif/core/blob/213e9203fd7f0f5aaffdcc9eb28d2828f7679373/src/config/config.ts#L475

but for the autocomplete refresh-hook use-case it doesn't work because the hook runs before oclif/core calls `plugins.delete`:

example with some logs to show order of execution

```
➜  plugin-plugins git:(cd/post-hooks) ✗ sf plugins uninstall @cristiand391/sf-plugin-fzf-cmp && unfunction _sf && autoload -U _sf
 ›   Warning: @oclif/plugin-autocomplete is a linked ESM module and cannot be auto-transpiled. Existing compiled source will be used instead.
 ›   Warning: @oclif/plugin-plugins is a linked ESM module and cannot be auto-transpiled. Existing compiled source will be used instead.
@salesforce/cli: Uninstalling @cristiand391/sf-plugin-fzf-cmp... done
running plugins:postuninstall hook
 ›   Warning: @oclif/plugin-autocomplete is a linked ESM module and cannot be auto-transpiled. Existing compiled source will be used instead.
done plugins:postuninstall hook
called `plugins.delete` from oclif/core
```

maybe we can remove the `plugins.delete` logic from oclif/core safely?
it was added here to avoid running postrun hooks from uninstalled plugins:
https://github.com/oclif/core/pull/805

Reply:
Yeah I think we can safely remove that once this is merged